File: backend/backend/Areas/Main/Models/CampaignNotes.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Areas.Main.Models;

public class CampaignNotes
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public int CampaignId { get; set; }
    public Campaign Campaign { get; set; }
    public int NoteId { get; set; }
    public Note Note { get; set; }
}

File: frontend/lib/models/campaign_notes.dart
import 'package:frontend/models/campaign.dart';
import 'package:frontend/models/note.dart';

class CampaignNotes {
  int? id;
  Campaign? campaign;
  Note? note;
  int? campaignId;
  int? noteId;

  CampaignNotes({
    this.id,
    this.campaign,
    this.note,
    this.campaignId,
    this.noteId
  });

  factory CampaignNotes.fromJson(Map<String, dynamic> json) {
    return CampaignNotes(
      id: json['id'],
      campaignId: json['campaignId'],
      campaign: json['campaign'] != null ? Campaign.fromJson(json['campaign']) : null,
      noteId: json['noteId'],
      note: json['note'] != null ? Note.fromJson(json['note']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'campaign': campaign,
      'note': note,
      'noteId': noteId,
      'campaignId': campaignId
    };
  }
}